### PR TITLE
Added CMake files

### PR DESCRIPTION
**Pull request by rajdakin**
As requeted, I'll create multiple pull request.

This one is adding the `CMakeLists.txt`, and ignoring `Makefile` (but you can always add it and commit it).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -193,6 +234,11 @@ Anyway, you need to have hardware 3D OpenGL support to run Blackvoxel.
 Windows
 -------
 
+Version 3.\*:
+You must use CMake to compile BlackVoxel.
+Unfortunately, I don't know how to do with Windows.
+
```
Comment (rajdakin) [inline README.md]: I let you modify this if you want.

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+unset(setto_CMAKE_LINKER_FLAGS)
+
+add_subdirectory(src/sc_Squirrel3)
+
+set(EXTRA_LIBS squirrel sqstdlib)
+
+add_executable(blackvoxel ${GAME_SOURCE_FILE})
+target_link_libraries(blackvoxel ${EXTRA_LIBS})
+set_target_properties(blackvoxel PROPERTIES COMPILE_DEFINITIONS "COMPILEOPTION_DATAFILESPATH=\"${CMAKE_INSTALL_PREFIX}\";COMPILEOPTION_SAVEFOLDERNAME=\"${COMPILEOPTION_SAVEFOLDERNAME}\";COMPILEOPTION_USEHOMEDIRSTORAGE=\"${COMPILEOPTION_USEHOMEDIRSTORAGE}\"")
```
Comment (rajdakin) [inline CMakeLists.txt]: I just see that `\"` is broken in some Make version (see https://cmake.org/cmake/help/v3.7/prop_tgt/COMPILE_DEFINITIONS.html)
The use of a Makefile is strongly recommanded then if you are in this case... (Hopefully you can always do `git add Makefile` or `git checkout Makefile`!)

**Comment by Blackvoxel** [comment]
Hi Radjdakin,

We sent you an email some days ago to explain how to do.

Did you received it?

**Comment by Blackvoxel** [comment]
See our email

**Pull request closed without merging**